components/tidb_query/src/rpn_expr/impl_op.rs

```diff
@@ -55,6 +56,33 @@ pub fn unary_not_decimal(arg: &Option<Decimal>) -> Result<Option<i64>> {
     Ok(arg.as_ref().map(|v| v.is_zero() as i64))
 }
 
+#[rpn_fn]
+#[inline]
+pub fn unary_minus_int(arg: &Option<Int>) -> Result<Option<Int>> {
+    match arg {
```

Comment:
Looks like you need separate implementations for unsigned / signed according to the current implementation and the TiDB implementation:

```rust
    pub fn unary_minus_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
        let val = try_opt!(self.children[0].eval_int(ctx, row));
        if self.children[0].is_unsigned() {
            let uval = val as u64;
            if uval > i64::MAX as u64 + 1 {
                return Err(Error::overflow("BIGINT", &format!("-{}", uval)));
            } else if uval == i64::MAX as u64 + 1 {
                return Ok(Some(i64::MIN));
            }
        } else if val == i64::MIN {
            return Err(Error::overflow("BIGINT", &format!("-{}", val)));
        }
        Ok(Some(-val))
    }
```

```go
func (b *builtinUnaryMinusIntSig) evalInt(row chunk.Row) (res int64, isNull bool, err error) {
	var val int64
	val, isNull, err = b.args[0].EvalInt(b.ctx, row)
	if err != nil || isNull {
		return val, isNull, err
	}

	if mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
		uval := uint64(val)
		if uval > uint64(-math.MinInt64) {
			return 0, false, types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("-%v", uval))
		} else if uval == uint64(-math.MinInt64) {
			return math.MinInt64, false, nil
		}
	} else if val == math.MinInt64 {
		return 0, false, types.ErrOverflow.GenWithStackByArgs("BIGINT", fmt.Sprintf("-%v", val))
	}
	return -val, false, nil
}
```

Reply:
PTAL again :)